### PR TITLE
support ChainerMN testing in pfnci

### PR DESCRIPTION
**Pull request by shmurai**
This commit adds
- scripts for running ChainerMN MPI tests on pfnci, and
- a script for building a Docker image used in the ChainerMN tests.

### CODE REVIEW

File: .pfnci/run.sh [hunk 1]
```diff
@@ -187,11 +187,55 @@ test_py27and35() {
   exit $((py27_test_status || py35_test_status))
 }
 
+# test_chainermn is a test function for chainermn
+test_chainermn() {
+  export PYENV_VERSION=""
+  . /root/.bash_profile
+
+  pyenv shell "2.7.16-cupy-${CUPY_VERSION}"
```
Comment (belldandyxtq) [inline .pfnci/run.sh]: Can we use list here?
```
TEST_CASES="2.7.16-cupy-${CUPY_VERSION} 3.5.7-cupy-${CUPY_VERSION} 3.7.3-cupy-${CUPY_VERSION}"
for TEST in $TEST_CASES
do
     pyenv shell $TEST
     test_chainermn_sub
done
```

Comment (belldandyxtq) [inline .pfnci/run.sh]: Can we use list here?
And we need to aggregate the return values

```
ret=0
TEST_CASES="2.7.16-cupy-${CUPY_VERSION} 3.5.7-cupy-${CUPY_VERSION} 3.7.3-cupy-${CUPY_VERSION}"
for TEST in $TEST_CASES
do
     pyenv shell $TEST
     test_chainermn_sub
     tmp_ret=$?
     ret=$(( ret || tmp_ret))
done
exit $ret
```

File: .pfnci/run.sh [hunk 2]
```diff
@@ -187,11 +187,55 @@ test_py27and35() {
...
+  pyenv shell "3.7.3-cupy-${CUPY_VERSION}"
+  test_chainermn_sub
+}
+
+# test_chainermn_sub runs tests for chainermn with current Python runtime
+test_chainermn_sub() {
+  marker='not slow'
+  if (( !GPU )); then
+    marker+=' and not gpu'
+    bucket=1
```
Comment (belldandyxtq) [inline .pfnci/run.sh]: Is this variable used?

File: .pfnci/script.sh [hunk 3]
```diff
@@ -56,6 +56,12 @@ main() {
           "asia.gcr.io/pfn-public-ci/chainer-ci-prep.${TARGET}" \
           bash /src/.pfnci/run.sh "${TARGET}"
       ;;
+	'chainermn' )
+      docker_args+=(--env="CUPY_VERSION=${CUPY_VERSION:-master}")
+      run "${docker_args[@]}" \
+          "asia.gcr.io/pfn-public-ci/chainermn-ci-prep-${CUDATAG:-cuda92}" \
```
Comment (belldandyxtq) [inline .pfnci/script.sh]: What do we do with 'cuda101' version?

Comment (shmurai) [reply .pfnci/script.sh]: We can specify the version of CUDA (currently, `cuda92` or `cuda101`) by `CUDATAG`. If not specified, `cuda92` is used as the default value.

File: .pfnci/run.sh (hunk 2 above)
Comment (shmurai) [reply .pfnci/run.sh]: Removed `bucket`

File: .pfnci/run.sh (hunk 1 above)
Comment (shmurai) [reply .pfnci/run.sh]: Fixed

File: .pfnci/chainermn-ci-prep.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+RUN touch $BASH_PROFILE && \
+    echo 'export PYENV_ROOT="$PYENV_ROOT"' >> $BASH_PROFILE && \
+	echo 'export PATH="$PYENV_ROOT/bin:$PATH"' >> $BASH_PROFILE && \
+	echo 'eval "$(pyenv init -)"' >> $BASH_PROFILE
+
+#
+# Install different versions of python and necessary packages
+#
+
+# Python 2.7.16
```
Comment (belldandyxtq) [inline .pfnci/chainermn-ci-prep.Dockerfile]: Would you remove also the installation of Python 2.7 from the docker, since we drop the Python 2.7 tests

Comment (shmurai) [reply .pfnci/chainermn-ci-prep.Dockerfile]: Removed. For flexibility and reduced code changes, the multi-version support using pyenv is kept.

File: .pfnci/chainermn-ci-prep.Dockerfile [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+RUN git clone https://github.com/pyenv/pyenv.git $PYENV_ROOT
+
+RUN touch $BASH_PROFILE && \
+    echo 'export PYENV_ROOT="$PYENV_ROOT"' >> $BASH_PROFILE && \
+	echo 'export PATH="$PYENV_ROOT/bin:$PATH"' >> $BASH_PROFILE && \
+	echo 'eval "$(pyenv init -)"' >> $BASH_PROFILE
+
+#
+# Install different versions of python and necessary packages
```
Comment (belldandyxtq) [inline .pfnci/chainermn-ci-prep.Dockerfile]: Some leftover

**Comment by belldandyxtq** [comment]
pfnci, test this please

**Comment by belldandyxtq** [comment]
pfnci, test this please

**Comment by belldandyxtq** [comment]
https://pypi.org/project/kerberos/

**Comment by belldandyxtq** [comment]
pfnci, test this please

**Pull request merged**